include/common/serialise_matrix.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+// BoB robotics includes
+#include "macros.h"
+
+// Third-party includes
+#include "third_party/path.h"
+
+// Eigen
+#include <Eigen/Core>
+
+// Standard C++ includes
+#include <fstream>
+
+namespace BoBRobotics {
+//! Read a matrix of type T from a binary file
+template<class T>
+auto
+readMatrix(const filesystem::path &filepath)
+{
+    // Open file
+    std::ifstream ifs;
+    ifs.exceptions(std::ios::badbit | std::ios::eofbit);
```

Comment:
what's the intention of this? it's a new unitialised stream so no flags should be set - checking ``ifs.good()`` after opening might be a good idea though - otherwise, I believe, it'll fail silently

Reply:
This sets up the exception mask for any *future* operations you do with ``ifs`` (including ``open()`` and ``read()``), otherwise they just fail silently (I just ran a little test to check this is true). I think the issue is that fstreams were added to C++ before exceptions were, so this was tacked on as an afterthought.

This does mean that we're not properly checking for errors elsewhere in BoB robotics, but I guess once a file is open then it's probably not too likely to fail on a ``read()`` or ``write()`` so it probs won't cause many problems.